From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: farrizdesignpro1339
Date: Sat, 27 Sep 2026 00:00:00 +0700
Subject: [PATCH 2/3] Add own product prop (ported as-is from LineageOS_gsi)

Ported from LineageOS_gsi 0006-Add-own-product-prop.patch
Generik, tidak ada dependensi Lineage, aman untuk PixelOS seventeen.
Dipakai via TARGET_PRODUCT_PROP += device/phh/treble/product.prop
(di custom_arm64.mk / rom.mk)

---
 product.prop | 50 ++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 50 insertions(+)
 create mode 100644 product.prop

diff --git a/product.prop b/product.prop
new file mode 100644
index 0000000..70f4953
--- /dev/null
+++ b/product.prop
@@ -0,0 +1,50 @@
+#Dalvik Hyperthread
+persist.sys.dalvik.multithread=true
+
+#Dalvik
+dalvik.vm.heapstartsize=16m
+dalvik.vm.heapgrowthlimit=320m
+dalvik.vm.heapsize=640m
+dalvik.vm.heapmaxfree=8m
+
+#Dexopt
+persist.bg.dexopt.enable=true
+pm.dexopt.bg-dexopt=everything
+dalvik.vm.dexopt-flags=v=n,o=a,u=y,m=y
+
+# IORap
+iorapd.readahead.enable=true
+ro.iorapd.enable=true
+
+# Render
+debug.composition.type=mdp
+persist.sys.composition.type=mdp
+debug.egl.force_msaa=0
+debug.egl.buffcount=4
+
+# Memory optimize
+persist.vendor.sys.memplus.enable=true
+
+# LMK
+ro.lmk.kill_timeout_ms=0
+ro.lmk.low=1001
+ro.lmk.psi_complete_stall_ms=700
+ro.lmk.psi_partial_stall_ms=70
+ro.lmk.thrashing_limit=100
+ro.lmk.thrashing_limit_decay=10
+ro.config.low_ram=false
+ro.lmk.use_minfree_levels=false
+ro.lmk.medium=900
+ro.lmk.critical_upgrade=false
+ro.lmk.enhance_batch_kill=false
+ro.lmk.enable_adaptive_lmk=false
+ro.lmk.use_minfree_levels=false
+ro.lmk.kill_heaviest_task=false
+ro.lmk.downgrade_pressure=100
+
+# Touch Improve
+windowsmgr.max_events_per_sec=90
+view.scroll_friction=0
+ro.min_pointer_dur=8
+debug.egl.hw=1
+video.accelerate.hw=1
--
2.54.0
